### DISCUSSION TITLE
install-php-extensions in a regulated environment

### DISCUSSION THREAD

**Discussion by adurand-ctie** [mlocati/docker-php-extension-installer#906] [Q&A] [answered]
Hi all,
I'm working in a regulated environment where my build machine doesn't have access to the Internet. Therefore we use Artifactory to be able to miror outside libraries and packages. I have defined a remote repository that points to https://pecl.php.net/packages.
I'm now trying to build a Docker image (based on frankenphp) and to retrieve PHP extensions with install-php-extensions.
Is there a way to tell the script to fetch the extensions from a different source, in my case my Artifactory mirror ? The extensions I need to install are apcu, intl, opcache, zip and pdo_pgsql.
Thanks in advance!
Adrien.

**Comment by mlocati** [comment, marked as answer]
Nope, there isn't a way to do that.

**Reply by adurand-ctie** [reply]
Hi Michele,
Thanks a lot for your fast answer. I'll look for a workaround.
Kind regards,
Adrien.

**Reply by mlocati** [reply]
A workaround could be something like:

1. use `install-php-extensions` on another machine that have access to pecl
2. take a note of these two lines:
   ```
   Packages to be kept after installation: <list1>
   Packages to be used only for installation:: <list2>
   ```
3. in the machine in the regulated enviromnent run:
   ```
   apt-get update -q
   apt-get install -qy <list1> <list2>
   pecl install <the extension you need>
   apt-get remove -qy <list2>
   ```
